src/helpers/update-user.js

```diff
@@ -2,21 +2,29 @@ const User = require('@lulibrary/lag-alma-utils/src/user')
 const ItemNotFoundError = require('@lulibrary/lag-utils/src/item-not-found-error')
 const sendToQueue = require('./send-to-queue')
 
-module.exports = (loanData, queueData) => {
-  const loanID = loanData.item_loan.loan_id
-  const userID = loanData.item_loan.user_id
-  const userCacheTable = process.env.UserCacheTableName
-  const eventUser = new User(userID, userCacheTable, process.env.AWS_REGION)
+// Helper method for SNS Handler Lambdas to add a new Loan to a User in the cache
+// or send the user ID to a message Queue for bulk updating if it is not in the cache
+module.exports = (userID, field, value, resourceData) => {
+  const eventUser = new User(userID, resourceData.userTable.name, resourceData.userTable.region)
 
   return eventUser.getData()
-    .then(() => {
-      return eventUser.addLoan(loanID).save()
-    })
+    .then(() => addValueToUser(eventUser, field, value))
     .catch(e => {
       if (e instanceof ItemNotFoundError) {
-        return sendToQueue(userID, queueData.name, queueData.owner)
+        return sendToQueue(userID, resourceData.userQueue.name, resourceData.userQueue.owner)
       } else {
         throw e
       }
     })
 }
+
+const addValueToUser = (user, field, value) => {
```

Comment:
I'm not sure an `addValueToUser` method makes sense takes a little effort to understand what's happening, `addLoan` and `addRequest` do make more sense.

Reply:
This was just to stop codeclimate complaining, if it's ok to ignore some issues it raises then I can change this back

Reply:
I agree it's an abstraction too far, given that the User class already provides an `add` method which does essentially the same thing, this just obscures the specific names of the `loan_ids` and `request_ids` fields on the User class. Again the only reason for the User#add method existing was to avoid complaints from codeclimate.

Reply:
Is the method on the user class `add` or `addLoan` ?